### ISSUE TITLE
Use Google Sign In API

### ISSUE THREAD

**Issue by friedger** [gdg-x/frisbee#616]
As described in here: https://developers.google.com/identity/sign-in/android/start-integrating

The google client needs the scope for G+

Games related code should be separated from normal use cases.
Handling Games API is out of scope for this issue.

The following parts need to be done:
- [ ] Use Auth Google Sign In api with G+ scope
- [ ] #526 (sign in button branding guide lines)
- [ ] Remove the second step (sign in) in onboarding
- [ ] #519 (show name of user)

**Comment by tasomaniac** [comment]
Should we use Firebase authentication? The API is really similar.

**Comment by friedger** [comment]
What is the advantage? We need the Google+ ID (for now).

We don't use firebase authentication on the hub. When we use firebase on the hub when can extend the Google SignIn to authenticate with firebase (https://firebase.google.com/docs/auth/android/google-signin).

The whole process is not very clear to me though...

**Comment by tasomaniac** [comment]
The new auth api is slightly better. Code will be cleaner. 
We don't have to do Firebase maybe. The advantage would be to store some info. It would be super easy and handy.  It can be things only care the Android app. Then if it is wanted hub can grab them in the future. 

We shouldn't wait for hub to do something for us anymore.

**Comment by friedger** [comment]
Firebase auth is out of scope for this issue.

**Comment by tasomaniac** [comment]
👍 My bad. Lets keep this issue simpler and focused.

**Comment by friedger** [comment]
started on branch https://github.com/gdg-x/frisbee/tree/refactor/auth

**Comment by friedger** [comment]
I started to work on this on branch feature/616_g_signin I didn't manage to finish it and leave it to somebody else to continue.

**Comment by tasomaniac** [comment]
I didn't close this to track problems with the latest changes. 
I've installed the alpha release on top of the prod release.
- The `silentSignIn` in `PlusUtils` didn't work immediately. Settings page still said that I was logged in. 
- I revoke the permission on Google Account settings on web. It didn't immediately log-out the user. I checked after half a day and I was logged out. 
  - Settings page was still showing that I was logged in. Quick solution to that would be to use Google Sign in button there as well. I would expect that the sign-in button will update itself. 
  - The old behavior on this case (revoking) was to ask again on every app screen until user either sign in again or cancel from the settings.
